File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/view/InheritanceMainView.java
package com.nerdscorner.mvplib.testapp.events.inheritance.view;

import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.nerdscorner.mvplib.events.view.BaseActivityView;
import com.nerdscorner.mvplib.testapp.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class InheritanceMainView extends BaseActivityView {
    @BindView(R.id.text) TextView textView;

    public InheritanceMainView(AppCompatActivity activity) {
        super(activity);
        ButterKnife.bind(this, activity);
    }

    @OnClick(R.id.some_button)
    public void onActionClicked() {
        getBus().post(new ActionClickedEvent());
    }

    public void setTextValue(CharSequence value) {
        textView.setText(value);
    }

    public static class ActionClickedEvent {
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/view/InheritanceMainView.kt
package com.nerdscorner.mvplib.testapp.events.inheritance.view

import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.nerdscorner.mvplib.events.view.BaseActivityView
import com.nerdscorner.mvplib.testapp.R

class InheritanceMainView(activity: AppCompatActivity) : BaseActivityView(activity) {
    @BindView(R.id.text)
    @JvmField
    var textView: TextView? = null

    init {
        ButterKnife.bind(this, activity)
    }

    @OnClick(R.id.some_button)
    fun onActionClicked() {
        bus.post(ActionClickedEvent())
    }

    fun setTextValue(value: CharSequence) {
        textView?.text = value
    }

    class ActionClickedEvent
}
